content/en/blog/_posts/2023-04-05-oci-security-profiles.md

```diff
@@ -0,0 +1,204 @@
+for the application logic on top. This way developers can focus on maintaining
+seccomp profiles which are way simpler and scoped to the application logic,
+without having a need to take the whole infrastructure setup into account.
+
+[runc]: https://github.com/opencontainers/runc
+[crun]: https://github.com/containers/crun
+
+But how to maintain those base profiles? For example, the amount of required
+syscalls for a runtime can change over its release cycle in the same way it can
+change for the main application. Base profiles have to be available in the same
+cluster, otherwise the main seccomp profile will fail to deploy. This means that
+they’re tightly coupled to the main application profiles, which acts against the
+main idea of base profiles. Distributing and managing them as plain files feels
+like an additional burden we have to solve.
+
+## OCI artifacts to the rescue
+
+The [latest release][spo-latest] of the Security Profiles Operator supports
+managing base profiles as OCI artifacts! Imagine OCI artifacts as lightweight
+container images, storing files in layers in the same way images do, but without
+a process to be executed. Those artifacts can be used to store security profiles
+like regular container images in compatible registries. This means they can be
+versioned, namespaced and annotated similar to regular container images.
+
+[spo-latest]: https://github.com/kubernetes-sigs/security-profiles-operator/releases/latest
+
+To see how that works in action, we have to specify a `baseProfileName` prefixed
+with `oci://` within a seccomp profile CRD, for example:
+
+```yaml
+apiVersion: security-profiles-operator.x-k8s.io/v1beta1
+kind: SeccompProfile
+metadata:
+  name: test
+spec:
+  defaultAction: SCMP_ACT_ERRNO
+  baseProfileName: oci://ghcr.io/security-profiles/runc:v1.1.5
+  syscalls:
+    - action: SCMP_ACT_ALLOW
+      names:
+        - uname
+```
+
+The operator will take care of pulling the content by using [oras][oras], as
+well as verifying the [sigstore (cosign)][cosign] signatures of the artifact. If
+the artifacts are not signed, then the SPO will reject them. The resulting
+profile `test` will then contain all base syscalls from the remote `runc`
+profile plus the additional allowed `uname` one. It is also possible to
+reference the base profile by its digest (SHA256) making the artifact to be
+pulled more specific, for example by referencing
+`oci://ghcr.io/security-profiles/runc@sha256:380…`.
+
+[oras]: https://oras.land
+[cosign]: https://github.com/sigstore/cosign
+
+The operator internally caches pulled artifacts up to 24 hours for 1000
+profiles, meaning that they will be refreshed after that time period, if the
+cache is full or the operator daemon gets restarted.
+
+Because the overall resulting syscalls are hidden to the user (we just have the
+`baseProfileName` in the CRD, not the syscalls itself), we additionally annotate
+the seccomp profile with the final `syscalls`:
```

Comment:
Try this:
```suggestion
Because the overall resulting syscalls are hidden from the user (I only have the
`baseProfileName` listed in the SeccompProfile, and not the syscalls themselves), I'll additionally
annotate that SeccompProfile with the final `syscalls`.

Here's how the SeccompProfile looks after I annotate it:
```

---

**Aside**
This is a private annotation - is this just a hint to the humans operating the workload?
If the annotation is intended to be machine readable / consumed elsewhere that's an architecture bug.
Kubernetes code should use public annotations (eg `spo.x-k8s.io/allowed-system-calls`).

Reply:
> This is a private annotation - is this just a hint to the humans operating the workload?

Yes, nothing machine readable.